src/services/completions.ts

```diff
@@ -1182,7 +1182,7 @@ namespace ts.Completions {
         function filterGlobalCompletion(symbols: Symbol[]): void {
             const isTypeOnly = isTypeOnlyCompletion();
             const allowTypes = isTypeOnly || !isContextTokenValueLocation(contextToken) && isPossiblyTypeArgumentPosition(contextToken, sourceFile, typeChecker);
-            if (isTypeOnly) {
+            if (allowTypes) {

```

Comment:
I don't think this is correct change. `KeywordFilters = KeywordCompletionFilters.TypeKeywords` would only offer type keywords? but that's not what you want here. I think you want isTypeOnly => `KeywordCompletionFilters.TypeKeywords`, otherwise if you have `allowTypes` then you want global + type keywords and rest of the time only global keywords ?

Reply:
After some discussion, we decided that isPossiblyTypeArgumentPosition is smart enough to rely on *only* providing types when it's true.

This simplifies code that relies on isPossiblyTypeArgumentPosition, as well as some tests. We no longer fall back to types+values, instead suggesting one or the other. isPossiblyTypeArgumentPosition should have a very low false positive rate since it relies on types, which means that people could see values-only completions a little more than before. The most common failure case I can think of is when a variable that should have a generic signature type actually has type `any`. I think this is acceptable.